modules/siddhi-extensions/math/src/test/java/org/wso2/siddhi/extension/math/PercentileFunctionExtensionTestCase.java

```diff
@@ -348,4 +738,134 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                eventArrived = true;
+                for (Event event : inEvents) {
+                    count++;
+                    switch (count) {
+                    case 1:
+                        Assert.assertEquals(10.0, event.getData(0));
+                        break;
+                    case 2:
+                        Assert.assertEquals(30.0, event.getData(0));
+                        break;
+                    case 3:
+                        Assert.assertEquals(100.0, event.getData(0));
+                        break;
+                    case 4:
+                        Assert.assertEquals(100.0, event.getData(0));
+                        break;
+                    case 5:
+                        Assert.assertEquals(100.0, event.getData(0));
+                        break;
+                    case 6:
+                        Assert.assertEquals(100.0, event.getData(0));
+                        break;
+                    case 7:
+                        Assert.assertEquals(100.0, event.getData(0));
+                        break;
+                    case 8:
+                        Assert.assertEquals(90.0, event.getData(0));
+                        break;
+                    case 9:
+                        Assert.assertEquals(90.0, event.getData(0));
+                        break;
+                    case 10:
+                        Assert.assertEquals(90.0, event.getData(0));
+                        break;
+                    default:
+                        Assert.fail();
+                    }
+                }
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+
+        inputHandler.send(new Object[] { 1, 10l });
+        inputHandler.send(new Object[] { 2, 30l });
+        inputHandler.send(new Object[] { 3, 100l });
+        inputHandler.send(new Object[] { 4, 40l });
+        inputHandler.send(new Object[] { 5, 80l });
+        inputHandler.send(new Object[] { 6, 60l });
+        inputHandler.send(new Object[] { 7, 20l });
+        inputHandler.send(new Object[] { 8, 90l });
+        inputHandler.send(new Object[] { 9, 70l });
+        inputHandler.send(new Object[] { 10, 50l });
+
+        Thread.sleep(100);
+        Assert.assertEquals(10, count);
+        Assert.assertTrue(eventArrived);
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void PercentileFunctionExtensionLong3() throws Exception {
+        logger.info("PercentileFunctionExtensionTestCase TestCase 12");
+
+        siddhiManager = new SiddhiManager();
+        String inStreamDefinition = "define stream inputStream (sensorId int, temperature long);";
```

Comment:
Thanks for pointing it out. Fixed with - https://github.com/wso2/siddhi/pull/192/commits/b50e2153220b2777e0cdcecf94bb5ab739d1f35e